Add Frostcrag Alchemy Sorter

charabaruk opened the pull request:
* Add plugin
  - To Locations - Overhauls section

* Add links
  - https://www.nexusmods.com/oblivion/mods/13782/ (original ver)
  - https://www.nexusmods.com/oblivion/mods/49969/ (updated ver)

* Add message to original FrostcragAlchemySorter.esp
  - Original version is obsolete.

* Add cleaning data
  - Original version
  - Updated version

charabaruk committed:
Add Frostcrag Alchemy Sorter

Includes cleaned and updated version which replaces the original.

sibir-ine commented on masterlist.yaml:
```diff
@@ -20395,6 +20395,26 @@ plugins:
+      - <<: *obsolete
+        type: error
+        subs: [ '[Frostcrag Alchemy Sorter Updated](https://www.nexusmods.com/oblivion/mods/49969)' ]
+        condition: 'not checksum("FrostcragAlchemySorter.esp", E855599E)'
```
The issue with using `not checksum()` is that if the authors of either version ever decide to update their versions, the message would trigger falsely for their users & cause confusion until we update it. I'd prefer to just explicitly check for the checksums of the old mod, even if it's dirty. However, I believe the author is @Sabre142 who often contributes here, so maybe they'd be willing to add version metadata to their version that could be used instead (see `version()` in the [docs](https://loot-api.readthedocs.io/en/stable/metadata/conditions.html#functions)).

Sabre142 replied:
I have added version info to the ESP's description field.

charabaruk replied:
@Sabre142 you are shiva7663? Or Dispensation?

Sabre142 replied:
I am Dispensation.

sibir-ine commented on masterlist.yaml:
```diff
@@ -20395,6 +20395,26 @@ plugins:
+      - link: 'https://www.nexusmods.com/oblivion/mods/49969/'
+        name: 'Frostcrag Alchemy Sorter Updated'
+    msg:
+      - <<: *obsolete
+        type: error
```
```suggestion
      - <<: *patchUpdateAvailable
        type: warn
```

`patchUpdateAvailable` would be better here as I don't believe this is an official update. An error message is also a bit excessive for issue that the mod fixes.

charabaruk replied:
I'm not sure if it would be, since Frostcrag Alchemy Sorter Updated isn't a patch but a replacement for Frostcrag Alchemy Sorter. The message for `obsolete` seems more appropriate, although even it isn't really accurate. Perhaps a custom message should be used here.

charabaruk replied:
I just went through a bunch of uses of `patchUpdateAvailable` and it seems to be overloaded with meanings -- in some cases being used for replacement mods like this, and in some places used to point out fix patches for the original mod to be applied on top.

I'll switch over to `patchUpdateAvailable` but that overloading of meaning should probably get cleaned up at some point too.

sibir-ine replied:
If there's really a case where the message causes confusion due to its ambiguity, we could reconsider it, but I'd like to avoid changing or adding messages unless absolutely necessary. Doing so means throwing out & requiring new translations, which is not a simple task as we don't have a team of translators for every language we support on standby.

sibir-ine commented on masterlist.yaml:
```diff
@@ -20395,6 +20395,26 @@ plugins:
+    msg:
+      - <<: *obsolete
+        type: error
+        subs: [ '[Frostcrag Alchemy Sorter Updated](https://www.nexusmods.com/oblivion/mods/49969)' ]
```
```suggestion
        subs: [ '[Frostcrag Alchemy Sorter Updated](https://www.nexusmods.com/oblivion/mods/49969/)' ]
```

charabaruk committed:
Update message for Frostcrag Alchemy Sorter

Switch message text and type/level. Normalize link URL in substitution.
Alter condition to specifically match CRC for original plugin.

charabaruk commented:
Updated message, including condition.

Sabre142 reviewed (changes requested):
Try utilising the new version metadata instead of a checksum for the condition:

```yaml
        condition: 'version("FrostcragAlchemySorter.esp", "1.0", <)'
```

charabaruk commented:
The condition would still have to include a checksum since Updated v1.0 as originally released does not include version in its description. So, `'version("FrostcragAlchemySorter.esp", "1.0" <) and not checksum("FrostcragAlchemySorter.esp", E855599E)'` would need to be the condition. Since it seems incredibly unlikely that the original will get updated (shiva7663 was last seen in 2017 and the original plugin was last updated ten years earlier), is it really worth that chance?

EDIT: On second thought, I'm overthinking it.

charabaruk committed:
Update message condition for Frostcrag Alchemy Sorter

Since Updated has been re-released with version in the esp description,
just check to see if there is no version or if it's less than 1.0.

charabaruk commented:
Condition updated.

sibir-ine commented on masterlist.yaml:
```diff
@@ -20395,6 +20395,26 @@ plugins:
+        util: '[TES4Edit v4.0.4](https://www.nexusmods.com/oblivion/mods/11536)'
+        itm: 2
+    clean:
+      - crc: 0xE855599E
```
```suggestion
      - crc: 0xBD3801E3
```

As the new updated version has a different checksum, the cleaning data should be updated.

charabaruk replied:
Could we keep both checksums? Especially given that the 1.0 re-release was just put out today.

sibir-ine replied:
It's just a clean checksum, so it's not really worth it. Besides, the message will encourage users of the previous version to update.

charabaruk replied:
Okay.

charabaruk committed:
Add checksum for Frostcrag Alchemy Sorter Updated 1.0 re-release

charabaruk committed:
Prune original Frostcrag Alchemy Sorter Updated 1.0 checksum

charabaruk commented:
Checksums updated.
